### PR TITLE
Creating a counter list

### PR DESCRIPTION
**Pull request by forki**

### CODE REVIEW

**Comment by MangelMaxime** [comment]
Can you please also remove this lines: https://github.com/forki/templates/blob/7701bf2bc977f4ba3f1ca31890817ae2c89741c2/src/react/Content/.editorconfig#L12-L14

It's will ensure people using `.editorconfig` to have 4 spaces indents and not 2 spaces.

**Comment by forki** [comment]
done

**Comment by MangelMaxime** [comment]
Thanks :)
I will review/test it this evening and merge it.

File: src/react/Content/src/State.fs [hunk 1]
```diff
@@ -4,43 +4,43 @@ open Elmish
...
-    urlUpdate result
-      { CurrentPage = Home
-        Counter = counter
-        CounterList = counterList
-        Home = home }
+    urlUpdate result (Home (Home.State.init()))
 
-let update msg model =
+let update msg (model:Model) =
     match msg with
```
Comment (et1975) [inline src/react/Content/src/State.fs]: I'd match on `msg, model` to avoid nested matching

Comment (forki) [inline src/react/Content/src/State.fs]: done. thx

**Comment by forki** [comment]
@et1975 more feedback?

**Comment by et1975** [comment]
I'll just quote what I said in the gitter
>There's no right or wrong way to define model or init.
Here's my mental model: init/update/view triple is a unit of modularity; think a mini-app, I should be able to run that module in isolation with a `Program`. App state is a tree that captures entire state of the app (if some module wants to init lazily it could return `None`).
It's a loosely-defined protocol, if you will. And the reason, as with all protocols, is encapsulation and integration. If you can keep in your head how each component behaves then, sure, optimize the hell out of it's api. The trade-off is - another person looking at your app will have to go look inside each component to understand how to interact with it.
I didn't want people new to elmish start with "optimal" mode of interaction between components, I wanted mini-app mental model to be the default.

You guys have a mental model that you agree on, so I'm not going to interfere.

**Comment by forki** [comment]
yes thanks for that comment. It's valuable. I think after the template split in demo and mini we can change that back in "mini"

**Comment by MangelMaxime** [comment]
@forki The template isn't compiling with your modifications. The changes you made to transform `Model` into `Page` breaks the main view function and the left menu resolution.

If you can fix it that would be nice otherwise I will need to revert some part of your job because I don't find a way to check if the current `menuItem` being resolve should be marked as active or not.

**Comment by forki** [comment]
Ok please revert that part. I'm not near a pc. Sorry

Am 10.06.2017 3:42 nachm. schrieb "Maxime Mangel" <notifications@github.com

**Comment by MangelMaxime** [comment]
No problem I will fix it and publish the template so.

**Comment by MangelMaxime** [comment]
@forki I published the template on [nuget](https://www.nuget.org/packages/Fable.Template.Elmish.React.Demo/0.1.0)

Can you please give it a try ?

**Pull request merged**